app/controllers/api/v1/admin/dashboard_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Api::V1::Admin::DashboardController < Api::V1::ApplicationController
+  before_action :authorize_admin
+
+  def index
+    orders = Order.where.not(status: :cart)
+              .where.not(created_at: nil)
+              .order(created_at: :desc)
+              .includes(:user)
+
+    pending_orders = orders.select { |order| order.status == "pending" }
+    completed_orders = orders.select { |order| order.status == "completed" }
+    delivered_orders = orders.select { |order| order.status == "delivered" }
+    @orders = pending_orders + completed_orders + delivered_orders
```

Comment:
I used this to select the orders by status, in order to display all the `pending `orders first. They are followed by `completed `and `delivered`.

Reply:
I see. In a large dataset, this could cause some memory / performance issues. You could order them from SQL directly.
Something like this should work:
```
orders..order("
  CASE
    WHEN status = 'pending' THEN 1
    WHEN status = 'completed' THEN 2
    WHEN status = 'delivered' THEN 3
  END
")
```

Reply:
I took your recommendation and updated the portion of code with SQL statements only in the API controller. Should I update the initial controller, too?